src/crud/document.py

```
@@ -32,6 +33,87 @@
+    now: datetime.datetime | None = None,
+) -> bool:
+    memory = _get_document_memory(document)
+    lifecycle = memory.get("lifecycle")
+    now_dt = now or datetime.datetime.now(datetime.timezone.utc)
+
+    def _parse(value: Any) -> datetime.datetime | None:
+        if not isinstance(value, str):
+            return None
+        try:
+            return parse_datetime_iso(value)
+        except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)